exir/backend/test/test_to_backend_multi_method.py

```diff
@@ -0,0 +1,517 @@
+            def __init__(self):
+                super().__init__()
+
+            def true_fn(self, x):
+                return torch.sin(x)
+
+            def false_fn(self, x):
+                return x + x
+
+            def forward(self, x):
+                return torch.cond(x > 0, self.true_fn, self.false_fn, [x])
+
+        sin_cos_edgeir_m = to_edge(
+            torch.export.export(SinCosModule(), (torch.ones(1),))
+        )
+        sin_add_edgeir_m = to_edge(
+            torch.export.export(SinAddModule(), (torch.ones(1),))
+        )
+
+        test_set = {
+            "sin_cos": (
+                sin_cos_edgeir_m.exported_program(),
+                BackendWithPreprocessAllPartitioner(),
+                [
+                    "FirstBackendWithPreprocessAll#4#aten.add.Tensor:#add:b'\\x00';",
+                    # True Module Partition
+                    "FirstBackendWithPreprocessAll#4#aten.sin.default:#sin:b'\\x01';",
+                    # False Module Partition
+                    "SecondBackendWithPreprocessAll#1#aten.cos.default:#cos:b'\\x03';",
+                ],
+            ),
+            "sin_add": (
+                sin_add_edgeir_m.exported_program(),
+                BackendWithPreprocessAllPartitioner(),
+                [
+                    # True Module Partition
+                    "FirstBackendWithPreprocessAll#4#aten.sin.default:#sin:b'\\x01';",
+                    # False Module Partition
+                    "FirstBackendWithPreprocessAll#4#aten.add.Tensor:#add:b'\\x00';",
+                ],
+            ),
+        }
+        self._test(test_set)
+
+    def test_multi_method_to_backend_not_found(self):
+        class SinModule(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+
+            def forward(self, x):
+                return torch.sin(x)
+
+        class AddModule(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+
+            def forward(self, x):
+                return x + x
+
+        sin_edgeir_m = to_edge(torch.export.export(SinModule(), (torch.ones(1),)))
+        add_edgeir_m = to_edge(torch.export.export(AddModule(), (torch.ones(1),)))
+        sin_partitioner = AllNodePartitioner(
+            "Invalid", [CompileSpec("sin", bytes([2]))]
+        )
+        add_partitioner = AllNodePartitioner(
+            "FirstBackendWithPreprocessAll", [CompileSpec("add", bytes([3]))]
+        )
+
+        test_set = {
+            "sin": (
+                sin_edgeir_m.exported_program(),
+                sin_partitioner,
+                [],
+            ),
+            "add": (
+                add_edgeir_m.exported_program(),
+                add_partitioner,
+                [],
+            ),
+        }
+        with self.assertRaisesRegex(
+            NotImplementedError, "Backend Invalid was not found."
+        ):
+            self._test(test_set)
+
+    def test_multi_method_end_to_end(self):
```

Comment:
@cccclai this should test the things you were asking about.

We go through to_backend_multimethod path, we lower to BackendWithCompilerDemo which doesn't have preprocess_multimethod implemented (which means it should take the default implementation path, the class method from the previous PR). 

We check that only one delegated blob is created when we are using processed bytes are the same, and we check that it runs end-to-end

Reply:
Yeah this looks nice, thank you!

Reply:
Maybe add this comment to the test case, so other people know the expectation for this test.